backend/plonk/bls12-377/marshal.go

```diff
@@ -349,6 +349,7 @@ func (vk *VerifyingKey) writeTo(w io.Writer, options ...func(*curve.Encoder)) (n
 		&vk.Kzg.G1,
 		&vk.Kzg.G2[0],
 		&vk.Kzg.G2[1],
+		&vk.Kzg.Lines,
```

Comment:
don't we need marshalling lines anymore? the out-circuit PLONK uses gnark-crypto KZG package which uses Lines, no? https://github.com/Consensys/gnark-crypto/blob/e77308824822cefa0bb066a9457eee0d426540ac/ecc/bls12-377/kzg/kzg.go#L247

Reply:
If I'm understanding correctly then you propose omitting Lines from PLONK verification key serialization/deserilization? I tried, but then the tests which check for structure stability after IO roundtrip, failed. In this PR we write the Lines when writing PLONK verification key (and also read when reading).